lib/livebook/session.ex

```diff
@@ -508,6 +513,11 @@ defmodule Livebook.Session do
     end
   end
 
+  defp schedule_memory_usage_update(state) do
+    ref = Process.send_after(self(), :memory_usage, 15000)
```

Comment:
This process runs as part of the Livebook app, but for code evaluation we use "runtime", which is generally a separate Elixir node (separate system process), so we should capture memory usage there instead. See the `Livebook.Runtime` protocol that defines the runtime contract. Currently all runtime types we have use `Livebook.Runtime.ErlDist.RuntimeServer` underneath, so that's the process where we want to capture memory usage. The RuntimeServer has an "owner" process, which is exactly the session, so we can send it `{:memory_usage, map()}` and document that in the Runtime protocol :)

Reply:
Ah, you are correctly capturing the memory by `:rpc`-ing the node, but my point is that it should be a part of the runtime contract, rather than session knowing about any node :)